build.gradle.kts

```diff
@@ -10,86 +9,115 @@ description = "A Result monad for modelling success or failure operations."
+// "jvmBenchmark" sourceSet we can access from kotlin mpp extension
+sourceSets.create("benchmark")
 
 benchmark {
     targets {
-        register("benchmark")
+        register("jvmBenchmark")
     }
 }
 
-dependencies {
-    implementation(kotlin("stdlib"))
-    testImplementation("junit:junit:4.13")
-    testImplementation(kotlin("test-common"))
-    testImplementation(kotlin("test-annotations-common"))
-    testImplementation(kotlin("test-junit"))
-    testImplementation(kotlin("test"))
+kotlin {
+    jvm {
+        withJava()
+    }
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                implementation(kotlin("stdlib-common"))
+            }
+        }
 
-    benchmarkImplementation(sourceSets.main.get().output + sourceSets.main.get().runtimeClasspath)
-    benchmarkImplementation("org.jetbrains.kotlinx:kotlinx.benchmark.runtime-jvm:0.2.0-dev-8")
-}
+        val commonTest by getting {
+            dependencies {
+                implementation(kotlin("test-common"))
+                implementation(kotlin("test-annotations-common"))
+            }
+        }
 
-tasks.withType<DependencyUpdatesTask> {
-    rejectVersionIf {
-        listOf("alpha", "beta", "rc", "cr", "m", "eap", "pr").any {
-            candidate.version.contains(it, ignoreCase = true)
+        val jvmMain by getting {
+            dependencies {
+                implementation(kotlin("stdlib-jdk8"))
+            }
         }
-    }
-}
 
-val dokka by tasks.existing(DokkaTask::class) {
-    outputFormat = "javadoc"
-    outputDirectory = "$buildDir/docs/javadoc"
-}
+        val jvmTest by getting {
+            dependencies {
+                implementation(kotlin("test-junit"))
+                implementation(kotlin("test"))
+            }
+        }
 
-val javadocJar by tasks.registering(Jar::class) {
-    group = LifecycleBasePlugin.BUILD_GROUP
-    description = "Assembles a jar archive containing the Javadoc API documentation."
-    archiveClassifier.set("javadoc")
-    dependsOn(dokka)
-    from(dokka.get().outputDirectory)
-}
+        val jvmBenchmark by getting {
+            dependsOn(jvmMain)
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx.benchmark.runtime-jvm:0.2.0-dev-8")
+            }
+        }
 
-val sourcesJar by tasks.registering(Jar::class) {
-    group = LifecycleBasePlugin.BUILD_GROUP
-    description = "Assembles a jar archive containing the main classes with sources."
-    archiveClassifier.set("sources")
-    from(sourceSets["main"].allSource)
+        all {
+            languageSettings.apply {
+                useExperimentalAnnotation("kotlin.contracts.ExperimentalContracts")
+            }
+        }
+
+    }
+
+    jvm().compilations.all {
+        kotlinOptions {
+            jvmTarget = "1.8"
+        }
+    }
+
+    jvm {
+        mavenPublication {
+            artifact(javadocJar.get())
+        }
+    }
```

Comment:
Can these two jvm blocks be merged?

Something like

```kotlin
    jvm {
        mavenPublication {
            artifact(javadocJar.get())
        }

        compilations.all {
            kotlinOptions.jvmTarget = "1.8"
        }
    }
```

Reply:
yes! ill move them into the same block as `withJava()`

Reply:
done